nvtabular/ops/data_schema.py

```diff
@@ -0,0 +1,173 @@
+#
+# Copyright (c) 2021, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+import logging
+import os
+
+import dask.dataframe as dd
+from nvtx import annotate
+
+from nvtabular.dispatch import DataFrameType
+
+from ..column import Columns
+from .base import Operator
+from .stat_operator import StatOperator
+
+LOG = logging.getLogger("nvtabular")
+
+
+class DataSchema(StatOperator):
+    SCHEMA_FILE_NAME = "schema.pb"
+
+    def __init__(self, tags_by_column, output_path=None):
+        super().__init__()
+        self.schema_path = os.path.join(output_path, self.SCHEMA_FILE_NAME) if output_path else None
+        self.col_names = []
+        self.col_types = []
+        self.col_dtypes = []
+        self.tags_by_column = tags_by_column
+        self.schema = None
+
+    @classmethod
+    def calculate_on_dataset(cls, dataset, tags_by_column, output_path=None, client=None):
```

Comment:
Since there's a convenience method that calls this on the `Workflow` class anyway, wonder if this method could be in-lined over there to avoid the additional imports here.

Reply:
Yeah, good point! I am not completely happy with the design of it as of now. Because going through the Workflow feels like a little bit of an overkill that also requires some boilerplate. Since what you typically want for both `DataSchema` & `DataStats` is to apply it on a `Dataset` where you use all columns in a `ColumnGroup`. Thinking out loud here, one other way we could do that is to overload the `>>` operator on the `Dataset` for that?